praw/models/reddit/rules.py

```diff
@@ -0,0 +1,296 @@
+        """Make a rule object from a data dict.
+
+        :param data: The dictionary of attributes and values
+        :returns: An instance of :class:`.Rule`.
+        """
+        return Rule(self._reddit, self.subreddit, _data=data)
+
+    def _make_rule_list(self, data_list: List[Dict[str, str]]) -> List[Rule]:
+        """Convert a list of rule dicts to Rule objects.
+
+        :param data_list: The list of dictionaries
+        :returns: A list of instances of :class:`.Rule`.
+        """
+        return [self._make_rule(ruledata) for ruledata in data_list]
+
+    def add(
+        self,
+        short_name: str,
+        kind: str,
+        description: str = "",
+        violation_reason: Optional[str] = None,
+    ) -> Rule:
+        """Add a removal reason to this subreddit.
+
+        :param short_name: The name of the rule.
+        :param kind: The kind of item that the rule applies to. One of
+            ``link``, ``comment``, or ``all``.
+        :param description: The description for the rule. Optional.
+        :param violation_reason: The reason that is shown on the report menu.
+            If a violation reason is not specified, the short name will be used
+            as the violation reason.
+        :returns: The Rule added.
+
+        To add rule ``No spam`` to the subreddit ``NAME`` try:
+
+        .. code-block:: python
+
+           reddit.subreddit('NAME').rules.add(
+               "No spam",
+               "all",
+               description="Do not spam. Spam bad")
+
+        """
+        data = {
+            "r": str(self.subreddit),
+            "description": description,
+            "kind": kind,
+            "short_name": short_name,
+            "violation_reason": short_name
+            if violation_reason is None
+            else violation_reason,
+        }
+        request_data = self._reddit.request(
+            "POST", API_PATH["add_subreddit_rule"], data=data
+        )
+        self._reddit._objector.objectify(request_data)
+        # We want any errors to be passed, but we don't want to use the
+        # objected value because it returns a LiveThread.
+        return self._make_rule(loads(request_data["json"]["data"]["rules"])[0])
+
+    def reorder(self, rule_list: List[Rule]) -> List[Rule]:
+        """Reorder the rules of a subreddit.
+
+        :param rule_list: The list of rules, in the wanted order. Each index of
+            the list indicates the position of the rule.
+        :returns: A list containing the rules in the specified order.
+
+        For example, to move the fourth rule to the first position, and then to
+        move the prior first rule to where the third rule originally was:
+
+        .. code-block:: python
+
+            subreddit = reddit.subreddit("subreddit")
+            rules = list(subreddit.rules)
+            new_rules = rules[3:4] + rules[1:3] + rules[0:1] + rules[4:]
+            # NOTE: Adding an upper bound returns a list
```

Comment:
I'm not sure what this comment means. It seems to be talking about the different between indexing (`lst[0]`) and slicing (`lst[0:1]`), and if so, the wording should be improved to be clearer. I'm not sure if it's even relevant at all here, as this is more of a Python thing than PRAW thing. But if you think it makes the example more readable to call that out, then I'm in favor of keeping a (rephrased) comment in.

Reply:
I wanted to comment why I'm using `lst[0:1]` as opposed to `[lst[0]]`, and that the alternate is a solution that uses more characters for the same thing. Maybe I can just drop the first comment altogether, and keep the alternate version.
```suggestion
```